back/app/models/concerns/participation_context.rb

```diff
@@ -7,183 +7,15 @@
-        validates :reacting_like_method, presence: true, inclusion: { in: REACTING_METHODS }
-        validates :reacting_dislike_enabled, inclusion: { in: [true, false] }
-        validates :reacting_dislike_method, presence: true, inclusion: { in: REACTING_METHODS }
-        validates :input_term, inclusion: { in: INPUT_TERMS }
-
-        before_validation :set_input_term
-      end
-      validates :ideas_order, inclusion: {
-        in: lambda do |pc|
-          Factory.instance.participation_method_for(pc).allowed_ideas_orders
-        end
-      }, allow_nil: true
-      validates :posting_limited_max, presence: true,
-        numericality: { only_integer: true, greater_than: 0 },
-        if: %i[can_contain_input? posting_limited?]
-      validates :reacting_like_limited_max, presence: true,
-        numericality: { only_integer: true, greater_than: 0 },
-        if: %i[can_contain_ideas? reacting_like_limited?]
-      validates :reacting_dislike_limited_max, presence: true,
-        numericality: { only_integer: true, greater_than: 0 },
-        if: %i[can_contain_ideas? reacting_dislike_limited?]
-      validates :allow_anonymous_participation, inclusion: { in: [true, false] }
-
-      # ideation?
-      with_options if: :ideation? do
-        validates :presentation_mode, presence: true
-      end
-
-      # voting?
-      with_options if: :voting? do
-        validates :voting_method, presence: true, inclusion: { in: VOTING_METHODS }
-        validate :validate_voting
-        validates :voting_term_singular_multiloc, multiloc: { presence: false }
-        validates :voting_term_plural_multiloc, multiloc: { presence: false }
-      end
-      validates :voting_min_total,
-        numericality: { greater_than_or_equal_to: 0, less_than_or_equal_to: :voting_max_total,
-                        if: %i[voting? voting_max_total],
-                        allow_nil: true }
-      validates :voting_max_total,
-        numericality: { greater_than_or_equal_to: :voting_min_total,
-                        if: %i[voting? voting_min_total],
-                        allow_nil: true }
-      validates :voting_max_votes_per_idea,
-        numericality: { greater_than_or_equal_to: 1, less_than_or_equal_to: :voting_max_total,
-                        if: %i[voting? voting_max_total],
-                        allow_nil: true }
-    end
-  end
-
-  def ideation?
-    participation_method == 'ideation'
-  end
-
-  def information?
-    participation_method == 'information'
-  end
-
-  def voting?
-    participation_method == 'voting'
-  end
-
-  def can_contain_ideas?
-    ideation? || voting?
-  end
-
-  def can_contain_input?
-    can_contain_ideas? || native_survey?
-  end
-
-  def posting_limited?
-    posting_method == 'limited'
-  end
-
-  def reacting_like_limited?
-    reacting_like_method == 'limited'
-  end
-
-  def reacting_dislike_limited?
-    reacting_dislike_method == 'limited'
   end
 
+  # TODO: JS - is this used?
```

Comment:
From looking through the code. I'm almost certain that this method isn't used anywhere.